client/src/router.js

```diff
@@ -134,4 +130,26 @@ export default new Router({
         //     component: () => import(/* webpackChunkName: "about" */ './views/WorkOrderStatuses.vue')
         // }
     ]
-})
\ No newline at end of file
+})
+appRouter.beforeEach((to, from, next) => {
+    // First check all possible blocks to being able to access the route
+    if (to.matched.some(record => record.meta.requiresAuth)) {  // Route requires logged in users
+        if (!store.getters.isLoggedIn) {
+            next('/login')
+            return
+        }
+    }
+
+    if (to.matched.some(record => record.meta.isUser)) {  // Route requires a specific username
+        let user = store.getters.user
+        if (to.meta.isUser != user.username) {
+            next('/products')
+            return
+        }
+    }
+
+    // If no reason is found to keep the user from the route, then allow them through
+    next()
+})
```

Comment:
The first tutorial you sent me follows this pattern of setting metadata on the routes and checking for it before processing a route. You of course don't need to follow that and can leave the beforeEnter pattern in place, but this felt clearer and cleaner.